src/rulesets/Base/arraymath.jl

```diff
@@ -19,6 +19,10 @@ end
 ##### `*`
 #####
 
+frule((_, Adot, Bdot), ::typeof(*), A, B) = A * B, muladd(Adot, B, A * Bdot)
+
+frule((_, Adot, Bdot, Cdot), ::typeof(*), A, B, C) = A*B*C, Adot*B*C + A*Bdot*C + A*B*Cdot
```

Comment:
Will anything go wrong if these are left without types?

Reply:
hmm, I mean it will mean the rule is hit rather than decomposing further.
But we can try without restrictions then if we run into bad cases add them later

Reply:
One bad thing about `Adot*B*C` is that, for 3 matrices, it will run the computation of how to group them 3 times. Without the rule, it would run once. But this ought to be tiny compared to matmul.

One good thing about it is that, for scalar-matrix-matrix, AD won't try to go inside the fused `mul!` implementation. Although perhaps nothing bad would happen there anyway.